developer/js/source/lexical-model-compiler/lexical-model-compiler.ts

```diff
@@ -0,0 +1,126 @@
+/*
+  lexical-model-compiler.ts: base file for lexical model compiler.
+*/
+
+/// <reference path="./lexical-model.ts" />
+/// <reference path="./model-info-file.ts" />
+
+import * as ts from "typescript";
+import * as fs from "fs";
+import * as path from "path";
+import { createTrieDataStructure } from "./build-trie";
+
+export default class LexicalModelCompiler {
+
+  /**
+   * Returns the generated code for the model that will ultimately be loaded by
+   * the LMLayer worker. This code contains all model parameters, and specifies
+   * word breakers and auxilary functions that may be required.
+   *
+   * @param model_id      The model ID. TODO: not sure if this is actually required!
+   * @param modelSource   A specification of the model to compile
+   * @param sourcePath    Where to find auxilary sources files
+   */
+  generateLexicalModelCode(model_id: string, modelSource: LexicalModelSource, sourcePath: string) {
+    let compiler = this;
+
+    // TODO: add metadata in comment
+    const filePrefix: string = `(function() {\n'use strict';\n`;
+    const fileSuffix: string = `})();`;
+    let func = filePrefix;
+
+    // Figure out what word breaker the model is using, if any.
+    let wordBreakerSpec = getWordBreakingSpec();
```

Comment:
Do we want to rename `getWordBreakingSpec` to `getWordBreakerSpec`? Also line 55.